examples/biobert_ner/bio_ner_predictor.py

```diff
@@ -105,58 +105,83 @@ def predict(self, data_batch: Dict[str, Dict[str, List[str]]]) \
+        """ Composes entity spans from subword entity predictions
+
+        Label Syntax:
+        A "B" label indicates the beginning of an entity, an "I" label
+        indicates the continuation of an entity, and an "O" label indicates
+        the absence of an entity.
+        Example: with - br - ##on - ##chi - ##oli - ##tis - .
+                 O    - B  - I    - I     - I     - I     - O
+
+        Due to possible instabilities of the model on out-of-distribution data,
+        sometimes the prediction may not follow the label format.
+        Example 1: with - br - ##on - ##chi - ##oli - ##tis - .
+                   O    - B  - I    - O     - I     - O     - O
+
+        Example 2: with - br - ##on - ##chi - ##oli - ##tis - .
+                   O    - O  - O    - I     - I     - I     - O
+
+        This method takes entity predictions of subwords and recovers the
+        set of complete entities, defined by the indices of their beginning
+        and ending subwords. (begin_idx, end_idx)
         """
-        entity_groups = []
-        entity_groups_disagg = []
+
+        complete_entities: List[Tuple[int, int]] = []
+        subword_entities: List[Dict[str, Any]] = []
 
         for entity in entities:
             subword = data_pack.get_entry(entity['tid'])
+
             if entity['label'] == 'B' and not subword.is_subword:
-                if entity_groups_disagg:
-                    entity_groups_disagg = \
-                        self._fill_sub_entities(entity_groups_disagg,
-                                                data_pack,
-                                                tids)
-                    entity_groups.append(entity_groups_disagg)
-                entity_groups_disagg = [entity]
+                # Flush the existing entity and start a new entity
+                if subword_entities:
+                    complete_entity = \
+                        self._complete_entity(subword_entities,
+                                              data_pack,
+                                              tids)
+                    complete_entities.append(complete_entity)
+                subword_entities = [entity]
             else:
-                entity_groups_disagg.append(entity)
+                # Continue accumuulating subword entities
+                subword_entities.append(entity)
 
-        if entity_groups_disagg:
-            entity_groups_disagg = self._fill_sub_entities(entity_groups_disagg,
-                                                           data_pack,
-                                                           tids)
-            entity_groups.append(entity_groups_disagg)
+        if subword_entities:
+            subword_entities = self._complete_entity(subword_entities,
+                                                     data_pack,
+                                                     tids)
+            complete_entities.append(subword_entities)
```

Comment:
So the `_complete_entity` function will try to include the whole word from the subword annotations. Is it possible that after these auto-expansion, we will have overlapping entities somehow?

Reply:
No it won't. `_complete_entity` will include `Subword` annotations until the first `Subword` that is a first segment, and the last `Subword` that's not a first segment. Whereas we always flush the `subword_entities` list when we see a "B" annotation that *is* a first segment. This means we'll never cross the line.